libafl_targets/src/cmps/mod.rs

```diff
@@ -390,8 +390,14 @@ impl CmpMap for CmpLogMap {
         } else {
             unsafe {
                 Some(CmpValues::Bytes((
-                    self.vals.routines[idx][execution].0.to_vec(),
-                    self.vals.routines[idx][execution].1.to_vec(),
+                    CmplogBytes::from_buf_and_len(
+                        self.vals.routines[idx][execution].0,
+                        CMPLOG_RTN_LEN as u8,
+                    ),
+                    CmplogBytes::from_buf_and_len(
```

Comment:
@tokatoka do we not support variable lengths here? Feels wrong

Reply:
wait this is libafl's cmpmap and it doesn't store lenght info.
so just keep the max length is the way

Reply:
Should it store the length info?

Reply:
Consider using a Box<[u8]> instead of [u8; 32]?

Reply:
That's the same as a vec, I wanted to get rid of allocations that's the whole point of this CL

Reply:
But, you can't, if you want to avoid the extra allocations. Variable length fields can't be a thing; if you want the smallest footprint to be ~18 bytes it's possible, but rn you're going to be hit with at least 40 bytes per instance.

Reply:
Potentially, you might be able to make a custom encoding with variable lengths, but then it won't be indexable, just iterable.

Reply:
Yes that's a tradeoff I don't understand, @tokatoka  will have to chime in

Reply:
no the memory footprint is not a problem we already allocate several hundreds of MB of cmplog already 😂

Reply:
Right, but maybe this is a good sign that it's about to get a lot worse :joy: